src/webgl/shaders/light_texture.frag

```diff
@@ -1,15 +1,24 @@
 precision mediump float;
 
 uniform vec4 uMaterialColor;
+uniform vec4 uTint;
 uniform sampler2D uSampler;
 uniform bool isTexture;
+uniform bool isTint;
 uniform bool uUseLighting;
 
 varying vec3 vLightWeighting;
 varying highp vec2 vVertTexCoord;
 
 void main(void) {
-  gl_FragColor = isTexture ? texture2D(uSampler, vVertTexCoord) : uMaterialColor;
+  if (isTexture) {
+    gl_FragColor = texture2D(uSampler, vVertTexCoord);  
+    if (isTint) 
+      gl_FragColor *= uTint / vec4(255, 255, 255, 255);
+  }
+  else
+    gl_FragColor = uMaterialColor;
+
```

Comment:
This whole section can be reduced to:
```
gl_FragColor = isTexture ? texture2D(uSampler, vVertTexCoord) * uTint : uMaterialColor;
```
with some data handling on the Renderer side. I would look into how fill is modified before being uploaded to the shader. Conditionals are expensive inside of a shader and should be avoided whenever possible. Luckily full-opacity white is equivalent to a null tint. We can leverage that and use [1, 1, 1, 1] to reduce complexity in the shader.